src/servers/grpc_server.cc

```diff
@@ -1606,17 +1606,39 @@ SharedMemoryControlHandler::Process(Handler::State* state, bool rpc_ok)
     TRTSERVER_Error* err = nullptr;
     if (request.has_register_()) {
       if (request.register_().has_system_shared_memory()) {
+        // system shared memory
         err = smb_manager_->Create(
             &smb, request.register_().name(),
             request.register_().system_shared_memory().shared_memory_key(),
-            request.register_().offset(), request.register_().byte_size());
+            nullptr, request.register_().system_shared_memory().offset(),
+            request.register_().byte_size(), TRTSERVER_MEMORY_CPU, 0);
+        if (err == nullptr) {
+          err = TRTSERVER_ServerRegisterSharedMemory(trtserver_.get(), smb);
+        }
       } else {
```

Comment:
check has_cuda_shared_memory.. the request could be improperly formed and not have either. You should have an error in that case.

Reply:
Done